### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by christopherlovell**
- Small fixes to Swift-EAGLE CAMELS load method.
- Creates a `log10metallicities` property on gas components.
- Fixes masking errors when using apertures, parametric_young_stars and other masks simultaneously 

## Issue Type
- Bug

## Checklist
- [x] I have read the [CONTRIBUTING.md]() -->
- [x] I have added docstrings to all methods
- [x] I have added sufficient comments to all lines
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no pep8 errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by WillJRoper** [comment]
Should we just move the log10 properties to the `Particles` base class?

File: src/synthesizer/particle/gas.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def __init__(
 
         # Set the metallicites and log10 equivalent
         self.metallicities = metallicities
-        self.log10metallicities = np.log10(self.metallicities)
+        # self.log10metallicities = np.log10(self.metallicities)
```
Comment (WillJRoper) [inline src/synthesizer/particle/gas.py:143]: Is there a reason this has been left?

File: src/synthesizer/load_data/load_camels.py [hunk 2]
```diff
@@ -676,6 +678,8 @@ def swifteagle_particle_assignment(
     ):
         gal = Galaxy()
         gal.redshift = redshift
+        gal.centre = pos[idx] * kpc
```
Comment (WillJRoper) [inline src/synthesizer/load_data/load_camels.py:681]: Just checking but definitely kpc here? The rest appears to be in Mpc

File: src/synthesizer/particle/stars.py [hunk 3]
```diff
@@ -612,6 +594,11 @@ def _aperture_mask(self, aperture_radius):
                 Radius of spherical aperture in kpc
         """
 
+        if self.centre is None:
+            raise ValueError(
+                "Centre of star particles must be set to use aperture mask."
+            )
+
         distance = np.sqrt(
```
Comment (WillJRoper) [inline src/synthesizer/particle/stars.py:602]: There's a property attached to all particles called something like `centered_coordinates` which automatically hands the centered coordinates and also ensures centre has been set at the same time. 

You could remove the check by using that property here.

**Review by WillJRoper — CHANGES_REQUESTED** [review]
A couple checks and one minor simplification.

File: src/synthesizer/particle/gas.py (hunk 1 above)
Comment (christopherlovell) [reply src/synthesizer/particle/gas.py]: No, removed

File: src/synthesizer/load_data/load_camels.py (hunk 2 above)
Comment (WillJRoper) [reply src/synthesizer/load_data/load_camels.py:681]: That's tragic 😂

File: src/synthesizer/particle/stars.py (hunk 3 above)
Comment (christopherlovell) [reply src/synthesizer/particle/stars.py]: moved to particle parent class, and simplified to use existing methods

**Review by WillJRoper — APPROVED** [review]

**Pull request merged**